notebooks/246-paint-by-example/246-paint-by-example.ipynb

```diff
@@ -0,0 +1,937 @@
+    "\n",
+    "This is the overall flow of the application:\n",
+    "![Flow Diagram](https://user-images.githubusercontent.com/103226580/236954918-f364b227-293c-4f78-a9bf-9dcebcb1034a.png)\n",
+    "\n",
+    "This is the detailed flowchart for the pipeline:\n",
+    "![pipeline-flowchart](https://github.com/openvinotoolkit/openvino_notebooks/assets/103226580/cde2d5c4-2540-4a45-ad9c-339f7a69459d)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install gradio\n",
+    "%pip install -q \"diffusers>=0.14.0\" openvino-dev \"transformers >= 4.25.1\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Download the model from [HuggingFace Paint-by-Example](https://huggingface.co/Fantasy-Studio/Paint-by-Example). This might take several minutes because it is over 5GB"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from diffusers import DPMSolverMultistepScheduler, DiffusionPipeline\n",
+    "\n",
+    "pipeline = DiffusionPipeline.from_pretrained(\"Fantasy-Studio/Paint-By-Example\")\n",
+    "\n",
+    "scheduler_inpaint = DPMSolverMultistepScheduler.from_config(pipeline.scheduler.config)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import gc\n",
+    "\n",
+    "extractor = pipeline.feature_extractor\n",
+    "image_encoder = pipeline.image_encoder\n",
+    "image_encoder.eval()\n",
+    "unet_inpaint = pipeline.unet\n",
+    "unet_inpaint.eval()\n",
+    "vae_inpaint = pipeline.vae\n",
+    "vae_inpaint.eval()\n",
+    "\n",
+    "del pipeline\n",
+    "gc.collect();"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Convert models to OpenVINO Intermediate representation (IR) format using ONNX\n",
+    "\n",
+    "Adapted from [236 Stable Diffusion v2 Infinite Zoom notebook](../236-stable-diffusion-v2/236-stable-diffusion-v2-infinite-zoom.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "import torch\n",
+    "import numpy as np\n",
+    "\n",
+    "model_dir = Path(\"model\")\n",
+    "model_dir.mkdir(exist_ok=True)\n",
+    "sd2_inpainting_model_dir = Path(\"model/paint_by_example\")\n",
+    "sd2_inpainting_model_dir.mkdir(exist_ok=True)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Functions to convert to ONNX and then to OpenVINO IR format"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convert_image_encoder_onnx(image_encoder: torch.nn.Module, onnx_path:Path):\n",
```

Comment:
Starting with [OpenVINO 2023.1](https://docs.openvino.ai/2023.1/openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html), converting models has been simplified with the introduction of the ov.convert_model function. This function allows for direct conversion to the OpenVINO IR model format, bypassing the previous need for ONNX to IR conversion. Following this, the ov.save_model function can be used to serialize the result of the conversion to disk.

For Instance:

```
import openvino as ov
from pathlib import Path
import torch

#Function to convert PyTorch model to OpenVINO IR format
def convert_model_to_ir(pytorch_model: torch.nn.Module, ir_path: Path, example_input, input_shape):
    pytorch_model.eval()
    with torch.no_grad():
        ov_model = ov.convert_model(pytorch_model, example_input=example_input, input=input_shape)
    ov.save_model(ov_model, ir_path)
    print(f'Model successfully converted to IR and saved to {ir_path}')

#Define paths for the converted models
IMAGE_ENCODER_IR_PATH_INPAINT = Path("model/paint_by_example/image_encoder.xml")

#Convert and save the model
example_input = torch.randn((1, 3, 224, 224))
convert_model_to_ir(image_encoder, IMAGE_ENCODER_IR_PATH_INPAINT, example_input, [(1, 3, 224, 224)])

```
The above is just an example. Also, please refer to the [236-stable-diffusion-v2-infinite-zoom.ipynb](https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/236-stable-diffusion-v2/236-stable-diffusion-v2-infinite-zoom.ipynb)

Reply:
Hi, @AnishaUdayakumar, 
@jlee17 has made the changes to the notebook to upgrade to openVINO 2023 and the PR has been updated, please let us know if there are any additional changes that need to be made